tools/eksDistroBuildToolingOpsTools/cmd/eksDistroOpsProwPlugin/server.go

```diff
@@ -0,0 +1,269 @@
+	labels []string
+	// Use prow to assign users to backported issue.
+	prowAssignments bool
+	// Allow anybody to do backports.
+	allowAll bool
+	// Create an issue on upstreampick conflict.
+	issueOnConflict bool
+	// Set a custom label prefix.
+	labelPrefix string
+
+	bare     *http.Client
+	patchURL string
+
+	repos   []github.Repo
+	mapLock sync.Mutex
+	lockMap map[upstreamPickRequest]*sync.Mutex
+}
+
+type upstreamPickRequest struct {
+	org    string
+	repo   string
+	issNum int
+}
+
+// ServeHTTP validates an incoming webhook and puts it into the event channel.
+func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	eventType, eventGUID, payload, ok, _ := github.ValidateWebhook(w, r, s.tokenGenerator)
+	if !ok {
+		return
+	}
+	fmt.Fprint(w, "Event received. Have a nice day.")
+
+	if err := s.handleEvent(eventType, eventGUID, payload); err != nil {
+		logrus.WithError(err).Error("Error parsing event.")
+	}
+}
+
+func (s *Server) handleEvent(eventType, eventGUID string, payload []byte) error {
+	l := logrus.WithFields(logrus.Fields{
+		"event-type":     eventType,
+		github.EventGUID: eventGUID,
+	})
+	switch eventType {
+	case "issues":
+		var ie github.IssueEvent
+		if err := json.Unmarshal(payload, &ie); err != nil {
+			return err
+		}
+		go func() {
+			if err := s.handleIssue(l, ie); err != nil {
+				s.log.WithError(err).WithFields(l.Data).Info("Issue creation failed.")
+			}
+		}()
+	default:
+		logrus.Debugf("skipping event of type %q", eventType)
+	}
+	return nil
+}
+
+func (s *Server) handleIssue(l *logrus.Entry, ie github.IssueEvent) error {
+	// Only consider newly opened issues and not PRs
+	if ie.Action != github.IssueActionOpened && !ie.Issue.IsPullRequest() {
+		return nil
+	}
+
+	org := ie.Repo.Owner.Login
+	repo := ie.Repo.Name
+	num := ie.Issue.Number
+	auth := ie.Sender.Name
+	title := ie.Issue.Title
+	body := ie.Issue.Body
+
+	// Do not create a new logger, its fields are re-used by the caller in case of errors
+	*l = *l.WithFields(logrus.Fields{
+		github.OrgLogField:  org,
+		github.RepoLogField: repo,
+		github.PrLogField:   num,
+	})
+
+	if auth != "eks-distro-bot" || !s.allowAll {
+		ok, err := s.ghc.IsMember(org, auth)
+		if err != nil {
+			return err
+		}
+		if !ok {
+			resp := fmt.Sprintf("only [%s](https://github.com/orgs/%s/people) org members may request may trigger automated issues. You can still create the issue manually.", org, org)
+			l.Info(resp)
+			return s.ghc.CreateComment(org, repo, num, resp)
+		}
+	}
+
+	//Currently handling Golang Patch Releases and Golang Minor Releases
+	var golangPatchReleaseRe = regexp.MustCompile(`(?m)^(?:Golang Patch Release:)\s+(.+)$`)
+	//var golangMinorReleaseRe = regexp.MustCompile(`(?m)^(?:Golang Minor Release:)\s+(.+)$`)
+
+	golangPatchMatches := golangPatchReleaseRe.FindAllStringSubmatch(ie.Issue.Title, -1)
+	if len(golangPatchMatches) != 0 {
+		if err := s.handle(l, ie.Issue.User.Login, &ie.Issue, org, repo, title, body, num); err != nil {
```

Comment:
should we rename `handle` to `handleGolangPatchReleaseEvent` or the like? so that it's clear that what it's doing....and I figure we'll have more handlers in the future as this grows a bit.